tests/unit/lib/upload_manager_test.py

```diff
@@ -1,66 +1,90 @@
+import tests.unit.azure_blob_mock  # noqa: F401
+
+import gzip
 import os
-from io import BytesIO
+import tarfile
 import tempfile
 import unittest
+import urllib
 
-from codalab.lib.upload_manager import UploadManager
-from codalab.worker.file_util import gzip_bytestring, remove_path, tar_gzip_directory
-
+from apache_beam.io.filesystem import CompressionTypes
+from apache_beam.io.filesystems import FileSystems
+from io import BytesIO
+from typing import IO, cast
 from unittest.mock import MagicMock
 from urllib.response import addinfourl
-import urllib
+
+from codalab.worker.file_util import gzip_bytestring, remove_path, tar_gzip_directory
+from tests.unit.server.bundle_manager import TestBase
 
 urlopen_real = urllib.request.urlopen
 
 
-class UploadManagerTest(unittest.TestCase):
-    def setUp(self):
-        class MockBundleStore(object):
-            def __init__(self, bundle_location):
-                self.bundle_location = bundle_location
+class UploadManagerTestBase(TestBase):
+    """A class that contains the base for an UploadManager test. Subclasses
+    can inherit from this class and unittest.TestCase and provide implementations
+    for the unimplemented methods in order to test different types of uploading.
+    """
 
-            def get_bundle_location(self, uuid):
-                return self.bundle_location
+    @property
+    def use_azure_blob_beta(self):
+        """Whether uploads use Azure Blob Storage."""
+        raise NotImplementedError
 
-        class MockBundleModel(object):
-            def update_bundle(self, *args, **kwargs):
-                return
+    def check_file_equals_string(self, file_subpath: str, expected_contents: str):
+        """Check that a file in the current bundle location has the specified string as its contents.
+        Args:
+            file_subpath (str): Subpath within the bundle. Set to an empty string if the bundle is just a single file and you want to specify just that file.
+            expected_contents (str): Expected string.
+        """
+        raise NotImplementedError
+
+    def listdir(self):
+        """List the files in the current bundle location."""
+        raise NotImplementedError
+
+    @property
+    def bundle_location(self):
+        """Get bundle location of the currently created bundle."""
+        return self.codalab_manager.bundle_store().get_bundle_location(self.bundle.uuid)
+
+    def setUp(self):
```

Comment:
is this camel case for a reason?
```suggestion
    def set_up(self):
```

Reply:
Yes, see https://docs.python.org/3/library/unittest.html#unittest.TestCase.setUp